src/lib/operators/aggregate_hash.cpp

```diff
@@ -333,6 +360,41 @@ KeysPerChunk<AggregateKey> AggregateHash::_partition_by_groupby_keys() const {
                 ++chunk_offset;
               });
             }
+
+            if constexpr (std::is_same_v<AggregateKey, AggregateKeyEntry>) {
```

Comment:
This is basically a check for a single-column groupby, right? Could you add this comment? I was looking for something like `groupby_columns.size()` and did not find it.

Reply:
Yes, will add it.

Reply:
Added:

> This shortcut only works if we are aggregating with a single GROUP BY column (i.e., when we use AggregateKeyEntry) - otherwise, we cannot establish a 1:1 mapping from keys_per_chunk to the result id.